scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -1289,7 +1289,7 @@ class FormatWriter(formatOps: FormatOps) {
             val beg = mods.lastOption.fold(tokens.after(ptokens.head)) { m =>
               tokens.next(tokens.tokenAfter(m))
             }
-            val useBody = b.eq(child) || p.eq(child)
+            val useBody = b.ne(null) && (b.eq(child) || p.eq(child))
```

Comment:
How is it possible that it's null actually? Shouldn't it be a placeholder or something?

Reply:
there are two reasons.

one, i added this trait to `Var` but since it uses `rhs: Option`, the body could be null. one option is to create a new tree that replaces null.

another is that, for some reason, the underscore in var is actually saved as None, maybe because semantically it's the same as without it.

Reply:
Should b be packed in an Option then? Just in case it's used in some other place?

Reply:
so we need to make two changes. first, fix the formatter (and for now, this pr is the fix), and then fix the parser and then adjust the formatter for the new version of the parser.

Reply:
Why not change the signature to Option[Tree] in the param of the unapply tuple?

For example here
https://github.com/scalameta/scalafmt/blob/cd07c999b377c6c7f0e75a75cec74a998ad8b3a2/scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala#L1255

Reply:
sure, i could do that. then we'd have to check body for null in this unapply anyway, and, in reality, since this is the only place this unapply is used, it doesn't seem to make a big difference.

perhaps the thing to do is to change `scalameta` asap, release and upgrade. then the formatter won't need any changes.

Reply:
scalameta/scalameta#3057